ethereumj-core/src/test/java/org/ethereum/solidity/CompilerTest.java

```diff
@@ -86,27 +88,87 @@ public void defaultFuncTest() throws IOException {
-        System.out.printf(contract.functions[0].toString());
+        System.out.print(contract.functions[0].toString());
     }
 
     @Test
     public void compileFilesTest() throws IOException {
 
-        File source = new File("src/test/resources/solidity/file1.sol");
+        Path source = Paths.get("src","test","resources","solidity","file1.sol");
 
-        SolidityCompiler.Result res = SolidityCompiler.compile(
-                source, true, ABI, BIN, INTERFACE, METADATA);
+        SolidityCompiler.Result res = SolidityCompiler.compile(source.toFile(), true, ABI, BIN, INTERFACE, METADATA);
         System.out.println("Out: '" + res.output + "'");
         System.out.println("Err: '" + res.errors + "'");
         CompilationResult result = CompilationResult.parse(res.output);
 
-        CompilationResult.ContractMetadata a = result.contracts.get("test1");
+        Assert.assertEquals("test1", result.getContractName());
+        Assert.assertEquals(source.toAbsolutePath(), result.getContractPath());
+
+        CompilationResult.ContractMetadata a = result.getContract(source, "test1");
         CallTransaction.Contract contract = new CallTransaction.Contract(a.abi);
-        System.out.printf(contract.functions[0].toString());
+        System.out.print(contract.functions[0].toString());
     }
 
+    @Test public void compileFilesWithImportTest() throws IOException {
+
+        Path source = Paths.get("src","test","resources","solidity","file2.sol");
+
+        SolidityCompiler.Result res = SolidityCompiler.compile(source.toFile(), true, ABI, BIN, INTERFACE, METADATA);
+        System.out.println("Out: '" + res.output + "'");
+        System.out.println("Err: '" + res.errors + "'");
+        CompilationResult result = CompilationResult.parse(res.output);
+
+        CompilationResult.ContractMetadata a = result.getContract(source, "test2");
+        CallTransaction.Contract contract = new CallTransaction.Contract(a.abi);
+        System.out.print(contract.functions[0].toString());
+    }
+
+    @Test public void compileFilesWithImportFromParentFileTest() throws IOException {
+
+        Path source = Paths.get("src","test","resources","solidity","foo","file3.sol");
+
+        SolidityCompiler.Option allowPathsOption = new SolidityCompiler.Options.AllowPaths(Collections.singletonList(source.getParent().getParent().toFile()));
+        SolidityCompiler.Result res = SolidityCompiler.compile(source.toFile(), true, ABI, BIN, INTERFACE, METADATA, allowPathsOption);
+        System.out.println("Out: '" + res.output + "'");
+        System.out.println("Err: '" + res.errors + "'");
+        CompilationResult result = CompilationResult.parse(res.output);
+
+        CompilationResult.ContractMetadata a = result.getContract(source, "test3");
+        CallTransaction.Contract contract = new CallTransaction.Contract(a.abi);
+        System.out.print(contract.functions[0].toString());
+    }
```

Comment:
could you assert here that there are 2 contracts totally and second is not null two and could be queried by name.